Src/IronPython.Modules/nt.cs

```diff
@@ -1549,16 +1549,26 @@ public class stat_result : IList, IList<object> {
         }
 
 #if FEATURE_PROCESS
+        [DllImport("libc", SetLastError = true, EntryPoint = "kill")]
+        private static extern int sys_kill(int pid, int sig);
+
         [Documentation(@"Send signal sig to the process pid. Constants for the specific signals available on the host platform 
 are defined in the signal module.")]
         public static void kill(CodeContext/*!*/ context, int pid, int sig) {
-            if (PythonSignal.NativeSignal.GenerateConsoleCtrlEvent((uint)sig, (uint)pid)) return;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ||
+                RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
+                if (sys_kill(pid, sig) == 0) return;
+
+                var error = Marshal.GetLastWin32Error();
+                throw PythonExceptions.CreateThrowable(PythonExceptions.OSError, error, strerror(error));
+            } else {
+                if (PythonSignal.NativeSignal.GenerateConsoleCtrlEvent((uint)sig, (uint)pid)) return;
 
-            //If the calls to GenerateConsoleCtrlEvent didn't work, simply
-            //forcefully kill the process.
-            Process toKill = Process.GetProcessById(pid);
-            toKill.Kill();
-            return;
+                // If the calls to GenerateConsoleCtrlEvent didn't work, simply
+                // forcefully kill the process.
+                Process toKill = Process.GetProcessById(pid);
+                toKill.Kill();
```

Comment:
Do we need to check for null here just in case? I've seen `GetProcessById` return null before, even when the process was alive. The user may pass in an invalid pid though a well.

Reply:
Do you remember on what platform it was returning `null`? I just checked the reference source for .NET Framework, .NET Core and Mono and it either throws or returns a `new Process` so it can't be `null`. Maybe you're thinking of something else?

Reply:
It was on .NET desktop, but it may have been several versions ago. I have checks in an app I wrote specifically for that reason.

Reply:
I just checked the .NET 4.5.1 reference source and it also can't return `null`. Maybe it was an older version which we don't support anymore.

Either way, I'm fine with this throwing if it's `null`. Better than silently not doing anything.